Source/PeleLM.cpp

```diff
@@ -6356,238 +6372,233 @@ PeleLM::compute_scalar_advection_fluxes_and_divergence (const MultiFab& Force,
-    typvals[Temp] = typical_values[Temp];
-    typvals[RhoH] = typical_values[RhoH];
-    for (int k = 0; k < nspecies; ++k) {
-       typvals[first_spec+k] = typical_values[first_spec+k]*typical_values[Density];
-    }
-    EB_set_covered_faces({D_DECL(EdgeState[0],EdgeState[1],EdgeState[2])},first_spec,nspecies,typvals);
-    EB_set_covered_faces({D_DECL(EdgeState[0],EdgeState[1],EdgeState[2])},Temp,1,typvals);
-    EB_set_covered_faces({D_DECL(EdgeState[0],EdgeState[1],EdgeState[2])},Density,1,typvals);
-    EB_set_covered_faces({D_DECL(EdgeState[0],EdgeState[1],EdgeState[2])},RhoH,1,typvals);
+  {
+     Vector<Real> typvals;
+     typvals.resize(NUM_STATE);
+     typvals[Density] = typical_values[Density];
+     typvals[Temp] = typical_values[Temp];
+     typvals[RhoH] = typical_values[RhoH];
+     for (int k = 0; k < nspecies; ++k) {
+        typvals[first_spec+k] = typical_values[first_spec+k]*typical_values[Density];
+     }
+     EB_set_covered_faces({D_DECL(EdgeState[0],EdgeState[1],EdgeState[2])},first_spec,NUM_SPECIES,typvals);
+     EB_set_covered_faces({D_DECL(EdgeState[0],EdgeState[1],EdgeState[2])},Temp,1,typvals);
+     EB_set_covered_faces({D_DECL(EdgeState[0],EdgeState[1],EdgeState[2])},Density,1,typvals);
+     EB_set_covered_faces({D_DECL(EdgeState[0],EdgeState[1],EdgeState[2])},RhoH,1,typvals);
   }
-
   EB_set_covered_faces({D_DECL(EdgeFlux[0],EdgeFlux[1],EdgeFlux[2])},0.);
  
 #else
- 
+   //////////////////////////////////////
+   //
+   // HERE IS THE NON-EB PROCEDURE
+   //
+   //////////////////////////////////////
 
-  //////////////////////////////////////
-  //
-  // HERE IS THE NON-EB PROCEDURE
-  //
-  //////////////////////////////////////
- 
- 
-  
-  // Initialize accumulation for rho = Sum(rho.Y)
-  for (int d=0; d<AMREX_SPACEDIM; d++) {
-    EdgeState[d]->setVal(0);
-    EdgeFlux[d]->setVal(0);
-  }
+   // Initialize accumulation for rho = Sum(rho.Y)
+   for (int dir=0; dir<AMREX_SPACEDIM; dir++) {
+      EdgeState[dir]->setVal(0.0);
+      EdgeFlux[dir]->setVal(0.0);
+   }
 
 #ifdef _OPENMP
 #pragma omp parallel if (Gpu::notInLaunchRegion())
 #endif
-  {
-    FArrayBox edgeflux[AMREX_SPACEDIM], edgestate[AMREX_SPACEDIM];
-    Vector<int> state_bc;
- 
-    for (MFIter S_mfi(Smf,TilingIfNotGPU()); S_mfi.isValid(); ++S_mfi)
-    {
-      const Box& bx = S_mfi.tilebox();
-      const FArrayBox& Sfab = Smf[S_mfi];
-      const FArrayBox& divu = DivU[S_mfi];
-      const FArrayBox& force = Force[S_mfi];
+   {
+      FArrayBox edgeflux[AMREX_SPACEDIM], edgestate[AMREX_SPACEDIM];
+      Vector<int> state_bc;
+      for (MFIter S_mfi(Smf,TilingIfNotGPU()); S_mfi.isValid(); ++S_mfi)
+      {
+         const Box& bx = S_mfi.tilebox();
+         const FArrayBox& Sfab = Smf[S_mfi];
+         const FArrayBox& divu = DivU[S_mfi];
+         const FArrayBox& force = Force[S_mfi];
 
 #ifdef AMREX_USE_CUDA
-      cudaError_t cuda_status = cudaSuccess;
+         cudaError_t cuda_status = cudaSuccess;
```

Comment:
This seems weird.  I'd probably initialize as bad and make a successful task set it as good.

Reply:
I'll admit, I don't catch the exceptions of cudaStreamSynchronize, I just needed a cuda_status to call that function. I can clean that up and catch any error.

Reply:
I replaced the cudaStreamSynchronize by Gpu::StreamSynchronize(), which is supposed to work for both Cuda and Hip as well as internally capture the error.